libs/custodian/regions.go

```diff
@@ -0,0 +1,173 @@
+package custodian
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/asaskevich/govalidator"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	appaws "github.com/brave-intl/bat-go/libs/aws"
+	"github.com/brave-intl/bat-go/libs/handlers"
+	"github.com/brave-intl/bat-go/libs/inputs"
+	"github.com/brave-intl/bat-go/libs/logging"
+)
+
+var (
+	custodianRegionsObj = "custodian-regions.json"
+	payoutStatusObj     = "payout-status.json"
+)
+
+// ExtractCustodianRegions - extract the custodian regions from the client
+func ExtractCustodianRegions(ctx context.Context, client appaws.S3GetObjectAPI, bucket string) (*CustodianRegions, error) {
+	logger := logging.Logger(ctx, "custodian.ExtractCustodianRegions")
+	// get the object with the client
+	out, err := client.GetObject(
+		ctx, &s3.GetObjectInput{
+			Bucket: &bucket,
+			Key:    &custodianRegionsObj,
+		})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get payout status: %w", err)
+	}
+	defer func() {
+		err := out.Body.Close()
+		logger.Error().Err(err).Msg("failed to close s3 result body")
```

Comment:
do we need a nil error check so log only errors

Reply:
resolved here: https://github.com/brave-intl/bat-go/pull/1569/commits/409ca0f45cd2c7b4cb6f7d8a1ee1ba7730a99bd3